pkg/kubelet/cadvisor/util_test.go

```diff
@@ -42,18 +45,57 @@ func TestCapacityFromMachineInfo(t *testing.T) {
 		},
 	}
 
-	// enable the features.HugePages
-	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.HugePages, true)()
+	hgFalseExpected := v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewMilliQuantity(int64(2000), resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(int64(2048), resource.BinarySI),
+	}
+	hgTrueExpected := v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewMilliQuantity(int64(2000), resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(int64(2048), resource.BinarySI),
+		"hugepages-5Ki":   *resource.NewQuantity(int64(51200), resource.BinarySI),
+	}
+	testCapacityFromMachineInfo(t, machineInfo, hgFalseExpected, false)
+	testCapacityFromMachineInfo(t, machineInfo, hgTrueExpected, true)
+}
+
+func testCapacityFromMachineInfo(t *testing.T, inputMI info.MachineInfo, expectedRL v1.ResourceList, fg bool) {
+	defer utilfeaturetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.HugePages, fg)()
+	actual := CapacityFromMachineInfo(&inputMI)
+	if !reflect.DeepEqual(actual, expectedRL) {
+		t.Errorf("when set hugepages %t, got resource list %v, want %v", fg, actual, expectedRL)
+	}
 
-	resourceList := CapacityFromMachineInfo(machineInfo)
+}
 
-	// assert the cpu and memory
-	assert.EqualValues(t, machineInfo.NumCores*1000, resourceList.Cpu().MilliValue(), "unexpected CPU value")
-	assert.EqualValues(t, machineInfo.MemoryCapacity, resourceList.Memory().Value(), "unexpected memory value")
+func TestCrioSocket(t *testing.T) {
+	assert.EqualValues(t, CrioSocket, crio.CrioSocket, "CrioSocket in this package must equal the one in github.com/google/cadvisor/container/crio/client.go")
+}
 
-	// assert the hugepage
-	hugePageKey := int64(5 * 1024)
-	value, found := resourceList[v1helper.HugePageResourceName(*resource.NewQuantity(hugePageKey, resource.BinarySI))]
-	assert.True(t, found, "hugepage not found")
-	assert.EqualValues(t, hugePageKey*10, value.Value(), "unexpected hugepage value")
+func TestUsingLegacyCadvisorStats(t *testing.T) {
+	testcases := []struct {
+		name            string
+		runtime         string
+		runtimeEndpoint string
+		expectedLinux   bool
+		expectedWindows bool
+	}{
+		{"case one: use legacy for both Linux and Windows", "docker", "/var/run/crio/crio.sock", true, true},
+		{"case two: use legacy for both Linux and Windows", "docker", "unix:///var/run/crio/crio.sock", true, true},
+		{"case three: use legacy for Linux, use non-legacy for Windows", "docker", "unix:///var/run/crio/fake.sock", true, false},
+		{"case four: use legacy for Linux, use non-legacy for Windows", "docker", "/var/run/crio/fake.sock", true, false},
+		{"case five: use non-legacy for both Linux and Windows", "rkt", "/var/run/crio/fake.sock", false, false},
+	}
+
+	for _, tc := range testcases {
+		ret := UsingLegacyCadvisorStats(tc.runtime, tc.runtimeEndpoint)
+		if goruntime.GOOS == "linux" {
```

Comment:
So IIUC, this test executes different paths based on whether it is being run on linux vs windows.  That seems a bit odd, in that we could get failures only when building on one OS.

Reply:
I'm not convinced this test case is useful given this.

Reply:
I think the "proper" way to do this would be to move this to one of the files separated by OS.  Maybe move it to `cadvisor_linux.go`, `cadvisor_windows.go`, and `cadvisor_unsupported.go`?  Then remove the goruntime.GOOS check depending on the case, and write unit tests for each.

Otherwise, I don't we shouldn't use goruntime.GOOS in the test.